pkg/products/monitoringcommon/alertmanagerConfig.go

```diff
@@ -0,0 +1,266 @@
+package monitoringcommon
+
+import (
+	"context"
+	"fmt"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/owner"
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	v1 "github.com/openshift/api/route/v1"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+const (
+	// alert manager configuration
+	alertManagerConfigSecretName            = "alertmanager-application-monitoring"
+	alertManagerConfigSecretFileName        = "alertmanager.yaml"
+	alertManagerEmailTemplateSecretFileName = "alertmanager-email-config.tmpl"
+	alertManagerConfigTemplatePath          = "alertmanager/alertmanager-application-monitoring.yaml"
+	alertManagerCustomTemplatePath          = "alertmanager/alertmanager-email-config.tmpl"
+
+	// Cluster infrastructure
+	clusterInfraName = "cluster"
+
+	// For Cluster ID
+	clusterVersionName = "version"
+
+	// For OpenShift console
+	openShiftConsoleRoute     = "console"
+	openShiftConsoleNamespace = "openshift-console"
+)
+
+func ReconcileAlertManagerSecrets(ctx context.Context, serverClient k8sclient.Client, installation *integreatlyv1alpha1.RHMI, productNamespace string, alertManagerRouteName string) (integreatlyv1alpha1.StatusPhase, error) {
```

Comment:
productNamespace (and if the above vars are moved, also alertManagerRouteName) should be able to come from the config object?

Reply:
In order to do this, we'd still need to pass the config object as a parameter since this function is in a different package than the products that use it (similar to how rhssocommon is implemented). Additionally, when this function is called by monitoring/reconciler.go `GetOperatorNamespace()` is passed but when the function is called by observability/reconciler.go `GetNamespace()` is passed. We could pass in a function instead but that seems like overkill. What do you think? @philbrookes @laurafitzgerald